app/controllers/trainees/personal_details_controller.rb

```diff
@@ -1,8 +1,40 @@
 module Trainees
   class PersonalDetailsController < ApplicationController
+    before_action :build_nationalities, only: %w[edit update]
```

Comment:
In the teacher training API we have this ADR which says we're going to rely more on memoisation rather than `before_action` hooks like this:

https://github.com/DFE-Digital/teacher-training-api/blob/master/docs/adr/0004-before-action-variables.md

Should we stick to the same approach here?

Reply:
Didn't we say we also wanted to try avoid using before actions? We could call `build_nationalities` from the edit/update methods

Reply:
🤔 Might of been before Davids time.

Reply:
done